src/main/java/io/cdap/plugin/gcp/common/CmekUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.common;
+
+import com.google.api.pathtemplate.ValidationException;
+import com.google.cloud.kms.v1.CryptoKeyName;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.common.base.Strings;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+
+import javax.annotation.Nullable;
+
+
+/**
+ *  Cmek Key Utility class to validate cmek in gcp plugins.
+ */
+public class CmekUtils {
+
+  /**
+   * This method validates the cmek formatted string.
+   *
+   * @param key cmek key formatted string
+   * @param collector  failure collector
+   * @return parsed CryptoKeyName object.
+   */
+  public static CryptoKeyName parseCmekKey(String key, FailureCollector collector) {
+    CryptoKeyName cmekKeyName = null;
+    try {
+      cmekKeyName = CryptoKeyName.parse(key);
+    } catch (ValidationException e) {
+      collector.addFailure(e.getMessage(), null)
+        .withConfigProperty(GCPConfig.NAME_CMEK_KEY).withStacktrace(e.getStackTrace());
+    }
+    return cmekKeyName;
+  }
+
+  /**
+   * This method validates the location of cmek key and bucket if it doesn't exists.
+   *
+   * @param storage GCS storage
+   * @param path the path of bucket to validate
+   * @param cmekKey the cmek key name
+   * @param location the location to create the bucket in if it doesn't exists
+   * @param collector failure collector
+   * @return true if bucket does not exist, otherwise false
```

Comment:
this is wrong, it also returns false if you can't tell if the bucket exists. Why does it need to return anything?